src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m

```diff
@@ -130,4 +133,59 @@ - (void)testClientStreamingRPC {
   [self waitForExpectationsWithTimeout:4. handler:nil];
 }
 
+- (void)testServerStreamingRPC {
+  __weak XCTestExpectation *expectation = [self expectationWithDescription:@"ServerStreaming"];
+  NSArray *expectedSizes = @[@31415, @9, @2653, @58979];
+  __block int index = 0;
+  id request = [[[[[[[RMTStreamingOutputCallRequestBuilder alloc] init]
+                    addResponseParameters:[[[[RMTResponseParametersBuilder alloc] init]
+                                            setSize:31415] build]]
+                   addResponseParameters:[[[[RMTResponseParametersBuilder alloc] init]
+                                           setSize:9] build]]
+                  addResponseParameters:[[[[RMTResponseParametersBuilder alloc] init]
+                                          setSize:2653] build]]
+                 addResponseParameters:[[[[RMTResponseParametersBuilder alloc] init]
+                                         setSize:58979] build]]
+                build];
+  [_service streamingOutputCallWithRequest:request handler:^(BOOL done, RMTStreamingOutputCallResponse *response, NSError *error){
+    XCTAssertNil(error, @"Finished with unexpected error: %@", error);
+    id expectedResponseBuilder = [[RMTStreamingOutputCallResponseBuilder alloc] init];
+    id expectedPayload = [[[[[RMTPayloadBuilder alloc] init]
+                            setType:RMTPayloadTypeCompressable]
+                           setBody:[NSMutableData dataWithLength:[expectedSizes[index] unsignedIntegerValue]]]
+                          build];
+    expectedResponseBuilder = [expectedResponseBuilder setPayload:expectedPayload];
+    id expectedResponse = [expectedResponseBuilder build];
+    XCTAssertEqualObjects(response, expectedResponse);
+    
+    [expectation fulfill];
+    index += 1;
+  }];
+  
+  [self waitForExpectationsWithTimeout:4 handler:nil];
+}
+
+- (void)testEmptyStreamRPC {
+  __weak XCTestExpectation *expectation = [self expectationWithDescription:@"EmptyStream"];
+  [_service fullDuplexCallWithRequestsWriter:[GRXWriter writerWithContainer:@[]]
+                                     handler:^(bool done, RMTStreamingOutputCallResponse *response, NSError *error) {
+                                       XCTAssertNil(error, @"Finished with unexpected error: %@", error);
+                                       XCTAssert(done, @"Unexpected response");
+                                       [expectation fulfill];
+                                     }];
+  [self waitForExpectationsWithTimeout:4 handler:nil];
+}
+
+- (void)testCancelAfterBeginRPC {
+  __weak XCTestExpectation *expectation = [self expectationWithDescription:@"CancelAfterBegin"];
+  ProtoRPC *call = [_service RPCToStreamingInputCallWithRequestsWriter:[GRXWriter writerWithContainer:@[]]
```

Comment:
But what we really want is a writer that never writes anything (nor finishes). If there isn't a constructor for that, we should add it too.